std/assembly/math.ts

```diff
@@ -1242,13 +1254,110 @@ function Rf(z: f32): f32 { // Rational approximation of (asin(x)-x)/x^3
 }
 
 @inline function expo2f(x: f32): f32 { // exp(x)/2 for x >= log(DBL_MAX)
-  const                        // see: musl/src/math/__expo2f.c
+  const                                // see: musl/src/math/__expo2f.c
     k    = <u32>235,
     kln2 = reinterpret<f32>(0x4322E3BC); // 0x1.45c778p+7f
   var scale = reinterpret<f32>(<u32>(0x7F + (k >> 1)) << 23);
   return NativeMathf.exp(x - kln2) * scale * scale;
 }
 
+@inline /** @internal */
+function pio2_large_quot(x: f32, u: i32): i32 {       // see: jdh8/metallic/blob/master/src/math/float/rem_pio2f.c
+  const coeff = reinterpret<f64>(0x3BF921FB54442D18); // π * 0x1p-65 = 8.51530395021638647334e-20
+  const bits = PIO2_TABLE;
+
+  var offset = (u >> 23) - 152;
+  var index  = offset >> 6;
+  var shift  = offset & 63;
+
+  var b0 = unchecked(bits[index + 0]);
+  var b1 = unchecked(bits[index + 1]);
```

Comment:
Wdyt of inlining the table as a switch? Probably just a crazy idea, but that'd then even work without a memory/allocator. Something along the lines of

```ts
function getPioVal(index: i32): u64 {
  switch (index) {
    default: assert(false);
    case 0: return 0xA2F9836E4E441529;
    case 1: return 0xFC2757D1F534DDC0;
    case 2: return 0xDB6295993C439041;
    case 3: return 0xFE5163ABDEBBC561;
  }
}
```

Reply:
It could be slower unfortunately. I tested similar approaches in wasm. But wasm hasn't computed jumps and use jump table instead which always slower than lookup tables

Reply:
Thought so, hmm. I wish we had a better way to put stuff into memory than having to use a full-blown gc'ed array.

Reply:
What if we had a special `Segment<T>` type that really just translates to a static memory segment? Let's say

```ts
const something: Segment<u64> = [1, 2, 3, 4];
...
load<u64>(changetype<usize>(something), immOffset);
```

Reply:
You worry about memory compactness? I think extra 8 or 16 bytes is not problem. Anyway usually wasm binaries compressed by gzip or brotli which greatly reduce any redundancy.

btw I build switch case approach and table for comparison final binary size:
 - switch/case `getPioVal` (227 bytes)
 - lookup table `getPioVal` (160 bytes)

Reply:
I see, yeah. Let's just do it with a table for now then :)